pageserver/src/metrics.rs

```diff
@@ -96,6 +96,16 @@ static CURRENT_PHYSICAL_SIZE: Lazy<UIntGaugeVec> = Lazy::new(|| {
     .expect("failed to define a metric")
 });
 
+#[cfg(not(test))]
```

Comment:
Why all the `cfg(not(test)) markers`? Is there any harm in having this metric in unit tests too?

Reply:
1. It's global state and tests run in parallel, we should avoid using global state to keep tests independent.
2. I didn't check but assumed that the metrics vec woudl implement Ownership model, i.e., that you can get only one metric from a metric vec for a given set of labels. That was wrong.

To keep things simple, I'll remote the `cfg(not(test))` and create a dummy metric in the one test that creates a RemoteTimelineClient.

They won't do any harm unless 
1. some concurrently running tests use the same label values and
2. those concurrently running tests read the metric value.